packages/ag-charts-community/src/chart/series/cartesian/lineUtil.ts

```diff
@@ -0,0 +1,402 @@
+        }
+
+        return { status: 'unknown' };
+    };
+    const fromFn = (marker: Marker, datum: LineNodeDatum, _status: NodeUpdateState) => {
+        const { status, point } = markerStatus(datum);
+        if (status === 'unknown') return { opacity: 0 };
+
+        const defaults = {
+            translationX: marker.translationX,
+            translationY: marker.translationY,
+            opacity: marker.opacity,
+            ...FROM_TO_MIXINS[status],
+        };
+
+        if (status === 'added') {
+            return { ...defaults, opacity: 0, translationX: point?.from?.x, translationY: point?.from?.y };
+        }
+
+        return defaults;
+    };
+
+    const toFn = (_marker: Marker, datum: LineNodeDatum, _status: NodeUpdateState) => {
+        const { status, point } = markerStatus(datum);
+        if (status === 'unknown') return { opacity: 0 };
+
+        const defaults = { translationX: datum.point.x, translationY: datum.point.y, opacity: 1 };
+
+        if (status === 'removed') {
+            return { ...defaults, translationX: point?.to?.x, translationY: point?.to?.y, opacity: 0 };
+        }
+
+        return defaults;
+    };
+
+    let status: NodeUpdateState = 'updated';
+    if (oldData.nodeData.length === 0 && newData.nodeData.length > 0) {
+        status = 'added';
+    } else if (oldData.nodeData.length > 0 && newData.nodeData.length === 0) {
+        status = 'removed';
+    }
+
+    const removePhaseFn = (ratio: number, path: Path) => {
+        if (status === 'added') addPhaseFn(1, path);
+        if (status === 'removed') render({ move: 0, out: 0 }, path);
+        if (status === 'updated') render({ move: 0, out: ratio }, path);
+    };
+    const updatePhaseFn = (ratio: number, path: Path) => {
+        if (status === 'added') addPhaseFn(1, path);
+        if (status === 'removed') removePhaseFn(0, path);
+        if (status === 'updated') render({ move: ratio }, path);
+    };
+    const addPhaseFn = (ratio: number, path: Path) => {
+        if (status === 'added') render({ move: 1, in: 1 }, path);
+        if (status === 'removed') removePhaseFn(0, path);
+        if (status === 'updated') render({ move: 1, in: ratio }, path);
+    };
+
+    const pathProperties = {
+        fromFn: (path: Path) => {
+            if (status === 'added') {
+                return { opacity: 0, ...FROM_TO_MIXINS['added'] };
+            }
+
+            return { opacity: path.opacity, ...FROM_TO_MIXINS[status] };
+        },
+        toFn: (path: Path) => {
+            if (status === 'added') {
+                return { opacity: 1, ...FROM_TO_MIXINS['added'] };
+            }
+
+            return { opacity: status === 'removed' ? 0 : path.opacity, ...FROM_TO_MIXINS[status] };
+        },
+    };
+
+    return { status, path: { addPhaseFn, updatePhaseFn, removePhaseFn }, pathProperties, marker: { fromFn, toFn } };
+}
+
+function findPointOnLine(a: { x: number; y: number }, b: { x: number; y: number }, targetX: number) {
```

Comment:
nit: Technically this function is no longer finding the point on the line, it is finding the y-point on the line given a fixed x. Probably worth changing the comment or renaming it so it doesn't get used in the wrong circumstance in the future.

Reply:
Updated